akka-persistence-typed/src/main/scala/akka/persistence/typed/scaladsl/EventSourcedBehavior.scala

```diff
@@ -13,6 +13,7 @@ import akka.actor.typed.internal.InterceptorImpl
 import akka.actor.typed.internal.LoggerClass
 import akka.actor.typed.scaladsl.ActorContext
 import akka.annotation.DoNotInherit
+import akka.persistence.Recovery
```

Comment:
Thanks for taking a first stab at this @an-tex 

I don't want to use classic Persistence classes in EventSourcedBehavior user API, and that `Recovery` has way too many ways of customizing recovery in ways that doesn't make sense or are not safe.

We should add a new Recovery here in Typed and only support the bare minimum that makes sense. I thought we agreed on one case in https://github.com/akka/akka/issues/25216#issuecomment-610970535